src/nvim/ops.h

```diff
@@ -62,6 +74,41 @@ enum GRegFlags {
   kGRegList    = 4   ///< Return list.
 };
 
+/// Definition of one register
+typedef struct yankreg {
+  char_u **y_array;  ///< Pointer to an array of line pointers.
+  linenr_T y_size;   ///< Number of lines in y_array.
+  char_u y_type;     ///< Register type: MLINE, MCHAR or MBLOCK.
+  colnr_T y_width;   ///< Register width (only valid for y_type == MBLOCK).
+  Timestamp timestamp;  ///< Time when register was last modified.
+  dict_T *additional_data;  ///< Additional data from ShaDa file.
+} yankreg_T;
+
+/// Convert register name into register index
+///
+/// @param[in]  regname  Register name.
+///
+/// @return Index in y_regs array or -1 if register name was not recognized.
+static inline int op_reg_index(const int regname)
+  FUNC_ATTR_CONST
+{
+  if (ascii_isdigit(regname)) {
```

Comment:
Note REAL_FATTR_PURE near the *declarations*. In this case declarations were written *explicitly*.